packages/interactivity/src/directives.js

```diff
@@ -87,6 +87,27 @@ export default () => {
 		} );
 	} );
 
+	const getGlobalEventDirective =
+		( type ) =>
+		( { directives, evaluate } ) => {
+			directives[ `on-${ type }` ]
+				.filter( ( { suffix } ) => suffix !== 'default' )
+				.forEach( ( entry ) => {
+					useEffect( () => {
+						const cb = ( event ) => evaluate( entry, event );
+						const globalVar = type === 'window' ? window : document;
+						globalVar.addEventListener( entry.suffix, cb );
+						return () =>
+							globalVar.removeEventListener( entry.suffix, cb );
+					}, [] );
+				} );
+		};
+
+	// data-wp-on-window--[event]
+	directive( 'on-window', getGlobalEventDirective( 'window' ) );
+	// data-wp-on-document--[event]
+	directive( 'on-document', getGlobalEventDirective( 'document' ) );
+
```

Comment:
Can we move this function outside the default export?

cc: @luisherranz @DAreRodz

Reply:
Same question applies to functions used in `data-wp-style`

```js
const newRule =
		/(?:([\u0080-\uFFFF\w-%@]+) *:? *([^{;]+?);|([^;}{]*?) *{)|(}\s*)/g;
	const ruleClean = /\/\*[^]*?\*\/|  +/g;
	const ruleNewline = /\n+/g;
	const empty = ' ';

	/**
	 * Convert a css style string into a object.
	 *
	 * Made by Cristian Bote (@cristianbote) for Goober.
	 * https://unpkg.com/browse/goober@2.1.13/src/core/astish.js
	 *
	 * @param {string} val CSS string.
	 * @return {Object} CSS object.
	 */
	const cssStringToObject = ( val ) => {
		const tree = [ {} ];
		let block, left;

		while ( ( block = newRule.exec( val.replace( ruleClean, '' ) ) ) ) {
			if ( block[ 4 ] ) {
				tree.shift();
			} else if ( block[ 3 ] ) {
				left = block[ 3 ].replace( ruleNewline, empty ).trim();
				tree.unshift( ( tree[ 0 ][ left ] = tree[ 0 ][ left ] || {} ) );
			} else {
				tree[ 0 ][ block[ 1 ] ] = block[ 2 ]
					.replace( ruleNewline, empty )
					.trim();
			}
		}

		return tree[ 0 ];
	};

Reply:
> Can we move this function outside the default export?

I don't understand the question 🙂

Reply:
I meant moving `getGlobalEventDirective`, `cssStringToObject` and the previous declarations to line 30. 
Before the export function. Is that possible?